tag-expressions/java/src/main/java/io/cucumber/tagexpressions/TagExpressionParser.java

```diff
@@ -21,15 +21,19 @@
         put("not", 2);
     }};
     private static final char ESCAPING_CHAR = '\\';
+    private String infix;
```

Comment:
I'd love to be rid of this bit of mutable state. 

Perhaps we can do this:

```
Expression exp = TagExpressionParser.parse("@food");
```

And then internally
```
public static Expression parse(String infix){
  return new TagExpressionParser(infix).parse();
}

Reply:
That would be nice but I needed the mutable state because the infix is needed in more than one function and I did not initially want to pass that around to multiple functions when something simple would work.  I could pass the `infix` string around but that seems not good as well.

Reply:
That's why I'm suggesting to create a tag parser object per infix that is parsed.